Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -339,8 +340,8 @@ void AODProducerWorkflowDPL::fillTrackTablesPerCollision(int collisionID,
             const double massZ = o2::track::PID::getMass2Z(trackPar.getPID());
             const double energy = sqrt((massZ * massZ) + (extraInfoHolder.tofExpMom * extraInfoHolder.tofExpMom));
             const double exp = extraInfoHolder.length * energy / (cSpeed * extraInfoHolder.tofExpMom);
-            extraInfoHolder.trackTime = static_cast<float>((tofMatch.getSignal() - exp) * 1e-3 - interactionTime); // tof time in \mus, FIXME: account for time of flight to R TOF
-            extraInfoHolder.trackTimeRes = 200e-3;                                                                 // FIXME: calculate actual resolution (if possible?)
+            extraInfoHolder.trackTime = static_cast<float>((tofSignal - exp) * 1e-3); // tof time in ns, taken from the definition of Ruben scaled by 1000 to convert from mus to ns to match the collisionTime
+            extraInfoHolder.trackTimeRes = 200e-3;                                    // FIXME: calculate actual resolution (if possible?)
```

Comment:
@shahor02 I am in favour of merging this. Unrelated: Don't we have an estimate of the track type already somewhere?

Reply:
@jgrosseo sorry, just saw the question. If you are asking whether we do reco-time PID: no, for that the TPC would need to provide a dedx -> PID algorithm.

Reply:
@shahor02 I wrote track type but mean track time resolution (sorry for this; and that's why I made the comment here). 
So do you already have a track time resolution (even course based on the type of track)?

Reply:
Not for ITS-TPC-TOF tracks: they are somewhat special since their time is given by the TOF signal stored in the match info (while the kinematic comes from ITS-TPC). In the https://github.com/AliceO2Group/AliceO2/blob/ce3e7457c3e9794d25d47ef156417b29f8a138a7/DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h#L169-L170 a 10 ns error is assigned (which is surely an overkill but is much more precise that the ITS-TPC time errors), but AOD producer does not use this method, since it relias on the vertex->tracks attachments tables.

Reply:
I think we should propagate these values into the AOD (even if you mark them as "overkill" for now). All this info should be available in the AOD converter, so it just requires adjusting the code, right?

Reply:
Yes, but I would do this in the opposite way: if TOF could provide a method to get an error for given {TOFCluster, track, LIntegral}, which could be used both in the RecoContainerCreateVariadic::create(...) and for the tracks directly accessed in the RecoContainer (like in the AODProducer, which would store it).

Reply:
* OK, who we should ping for this?
* In addition, how do we fill `trackTimeRes` for non TOF tracks?

Reply:
What kind of TOF error do you mean? The time resolution for PID?

Reply:
@jgrosseo for tracks with TRD the error corresponds to the nominal triggered BC duration (i.e. a few 100 ps), for ITS-TPC tracks: uncertainty in matching point Z / vdrift.

@noferini : essentially, the uncertainty on the expected time.

Reply:
For uncertainties on the expected times we need a tuned parameterization consistent with data and MC.
In run-2 it was encoded in a base class + OADB. I guess we will do the same for Run-3 (OADB -> CCDB).
If I'm not wrong @njacazio started to work on that

Reply:
TrackTime contains here the track time estimate based on TOF (absolute wrt BC). We do *not* talk about the estimated time with a momentum and particle hypothesis. Do we talk about the same thing?

Reply:
> @jgrosseo for tracks with TRD the error corresponds to the nominal triggered BC duration (i.e. a few 100 ps), for ITS-TPC tracks: uncertainty in matching point Z / vdrift.

Can we add this to the AOD producer already?

Reply:
I see. A deltat (t_tof - t_track) is avaliable with the matching info.
For the resolution we depend on the track time precision which is available with the track.
Is this what you need?

Reply:
@jgrosseo 

> We do not talk about the estimated time with a momentum and particle hypothesis. Do we talk about the same thing?

Does this matter if we speak about the uncertainty? 

> Can we add this to the AOD producer already?

Do you mean to fill `extraInfoHolder.trackTime` and `extraInfoHolder.trackTimeRes` for all tracks types?

Reply:
> Do you mean to fill extraInfoHolder.trackTime and extraInfoHolder.trackTimeRes for all tracks types?
Yes!